### PR TITLE
feat(cdk-plugin): add Serverless construct static export

### PR DESCRIPTION
**Pull request by Eikix**
A new version of the serverless cdk plugin that does not utilize the "exports" syntax of the package.json file.

### CODE REVIEW

File: backend/orchestrator/package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
   },
   "dependencies": {
     "@swarmion/orchestrator-contracts": "^0.8.8",
+    "@swarmion/serverless-cdk-plugin": "0.8.8",
```
Comment (GuillaumeLagrange) [inline backend/orchestrator/package.json]: carret here to stay consistent

File: backend/orchestrator/resources/dynamodb.ts [hunk 2]
```diff
@@ -1,14 +1,17 @@
 import { AttributeType, BillingMode, Table } from 'aws-cdk-lib/aws-dynamodb';
 import { Construct } from 'constructs';
 
+import type { ServerlessProps } from '@swarmion/serverless-cdk-plugin';
```
Comment (GuillaumeLagrange) [inline backend/orchestrator/resources/dynamodb.ts:4]: Does this work ?
```typescript
import { ServerlessProps, getCdkPropertyHelper, ServerlessConstruct } from '@swarmion/serverless-cdk-plugin';
```

Comment (GuillaumeLagrange) [reply backend/orchestrator/resources/dynamodb.ts:4]: If it doesn't we should go with

```typescript
import ServerlessCdkPlugin, { ServerlessProps } from '@swarmion/serverless-cdk-plugin';
```

File: backend/orchestrator/serverless.ts [hunk 3]
```diff
@@ -11,18 +12,20 @@ import {
...
   frameworkVersion,
   configValidationMode: 'error',
   plugins: [
     'serverless-esbuild',
+    '@swarmion/serverless-cdk-plugin',
     '@swarmion/serverless-plugin',
     'serverless-iam-roles-per-function',
     'serverless-analyze-bundle-plugin',
   ],
+  serverlessConstruct: OrchestratorDynamodb,
```
Comment (GuillaumeLagrange) [inline backend/orchestrator/serverless.ts:28]: rename this key `construct` in `CdkPluginConfig`

File: packages/serverless-cdk-plugin/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+### Use-case examples
+
+<img width="500" height="500" src="./static/lambda-sqs-lambda.drawio.svg"/>
+
+#### Code example:
+
+Here's a short code snippet to demonstrate how to use the plugin. Code snippets are taken from the [integrationTests](./integrationTests/) folder.
+
+Typically you would create a construct in the desired folder. For instance, a DynamoDB construct.
```
Comment (GuillaumeLagrange) [inline packages/serverless-cdk-plugin/README.md]: from the old PR by @adriencaccia 

We shouldn't use 'you' and keep everything third person

+ remove typically .... would formulation to be more assertive

File: packages/serverless-cdk-plugin/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      billingMode: BillingMode.PAY_PER_REQUEST,
+    });
+
+    this.dynamodbArn = tableArn;
+    this.dynamodbName = tableName;
+  }
+}
+```
+
+Then, you would pass the construct to the `serverless.ts` file at the serverlessConstruct key:
```
Comment (GuillaumeLagrange) [inline packages/serverless-cdk-plugin/README.md]: third person

File: packages/serverless-cdk-plugin/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  plugins: ['../../src'],
+  provider: {
+    name: 'aws',
+  },
+  // Other properties ...
+  serverlessConstruct: MyConstruct,
+};
+```
+
+And that's it! You can now reference your DynamoDB in your Serverless framework code seamlessly (e.g. in your lambdas).
```
Comment (GuillaumeLagrange) [inline packages/serverless-cdk-plugin/README.md]: and also here third person :)

File: packages/serverless-cdk-plugin/README.md [hunk 7]
```diff
@@ -59,7 +59,7 @@ export class MyConstruct extends Construct {
 }
```
Comment (GuillaumeLagrange) [inline packages/serverless-cdk-plugin/README.md:59]: squash this commit with the previous commit

File: backend/orchestrator/package.json (hunk 1 above)
Comment (GuillaumeLagrange) [inline backend/orchestrator/package.json]: carret here

File: packages/serverless-cdk-plugin/src/types/cdkPluginConfig.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
-import { Construct } from 'constructs';
+import { ServerlessConstruct } from './serverlessConstruct';
 
 export type CdkPluginConfig = {
```
Comment (GuillaumeLagrange) [inline packages/serverless-cdk-plugin/src/types/cdkPluginConfig.ts:3]: open question: shouldn't we rename this to `ServerlessCdkPluginConfig` to be more in line with the plugin name ? Not sure what's the convention here @CorentinDoue

File: packages/serverless-cdk-plugin/static/lambda-sqs-lambda.drawio.svg [hunk 9]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (GuillaumeLagrange) [inline packages/serverless-cdk-plugin/static/lambda-sqs-lambda.drawio.svg]: Make the name clearer (comment from the old PR)

File: packages/serverless-cdk-plugin/src/types/cdkPluginConfig.ts [hunk 10]
```diff
@@ -1,5 +1,5 @@
...
-  cdkConstruct: typeof Construct;
+  serverlessConstruct: typeof ServerlessConstruct;
```
Comment (GuillaumeLagrange) [inline packages/serverless-cdk-plugin/src/types/cdkPluginConfig.ts:4]: we want to keep this as a Construct imo. And dynamically check if it's a serverless Construct to give the props

File: packages/serverless-cdk-plugin/src/serverlessCdk.ts [hunk 11]
```diff
@@ -122,12 +141,11 @@ export class ServerlessCdkPlugin implements Plugin {
...
     const serverlessConfigFile = await getServerlessConfigFile();
-    const MyConstruct = serverlessConfigFile.cdkConstruct;
+    const ServerlessCdkConstruct = serverlessConfigFile.serverlessConstruct;
 
-    this.construct = new MyConstruct(
-      this.stack,
-      'serverlessCdkBridgeConstruct',
-    );
+    this.construct = new ServerlessCdkConstruct(this.stack, 'cdk', {
+      serverless: serverlessConfigFile,
```
Comment (GuillaumeLagrange) [inline packages/serverless-cdk-plugin/src/serverlessCdk.ts:147]: dynamically check if we have a `ServerlessConstruct` and only give the props if we have. We want to stay compatible with simple `Construct`

**Review by GuillaumeLagrange — CHANGES_REQUESTED** [review]

File: packages/serverless-cdk-plugin/integrationTests/defaultUsage/serverless.ts [hunk 12]
```diff
@@ -13,8 +17,11 @@ const serverlessConfiguration: AWS & CdkPluginConfig = {
   functions: {
     testFunction: {
       environment: {
-        ORCHESTRATOR_TABLE_NAME: '${serverlessCdkBridgePlugin:dynamodbName}',
-        ORCHESTRATOR_TABLE_ARN: '${serverlessCdkBridgePlugin:dynamodbArn}',
+        ORCHESTRATOR_TABLE_NAME: getCdkProperty('dynamodbName'),
+        ORCHESTRATOR_TABLE_ARN: getCdkProperty('dynamodbArn'),
+        TEST_OUTPUT_SERVERLESS_CONFIG_VALUE: getCdkProperty(
+          'testServerlessConfigValue',
+        ),
```
Comment (CorentinDoue) [inline packages/serverless-cdk-plugin/integrationTests/defaultUsage/serverless.ts:24]: c'est quoi ça ?

Comment (Eikix) [reply packages/serverless-cdk-plugin/integrationTests/defaultUsage/serverless.ts:24]: C'est pour tester si le construct cdk a accès à des valeurs internes au sls config file

**Review by GuillaumeLagrange — APPROVED** [review]
🚀🚀🚀🚀

**Comment by GuillaumeLagrange** [comment]
@adriencaccia or @fargito this time we're ready 👀

Sorry for all the different PRs on this subject

**Review by fargito — APPROVED** [review]
Great work!

**Pull request merged**